src/main.rs

```diff
@@ -240,6 +271,129 @@ async fn main() -> Result<()> {
+                        desired_filename.push(filename);
+                        desired_filename
+                    } else {
+                        desired
+                    }
+                }
+                // User didn't specify an output path. Add ".iron" to the input path.
+                None => {
+                    let mut output_path = infile;
+                    let new_extension = match output_path.extension() {
+                        Some(extension) => {
+                            let mut ext_os = extension.to_os_string();
+                            ext_os.push(".iron");
+                            ext_os
+                        }
+                        None => std::ffi::OsString::from("iron"),
+                    };
+                    output_path.set_extension(new_extension);
+                    output_path
+                }
+            };
+            let mut users_or_groups = group_ids
+                .iter()
+                .map(|group| group.into())
+                .collect::<Vec<UserOrGroup>>();
+            let mut users = user_ids
+                .iter()
+                .map(|user| user.into())
+                .collect::<Vec<UserOrGroup>>();
+            users_or_groups.append(&mut users);
+            let grants = ExplicitGrant::new(true, &users_or_groups);
+            let opts = DocumentEncryptOpts::new(None, None, EitherOrBoth::Left(grants));
+            let encrypt_result = sdk.document_encrypt(&file, &opts).await?;
+            let successes = encrypt_result
+                .grants()
+                .iter()
+                .map(|u_or_g| match u_or_g {
+                    UserOrGroup::User { id } => format!("User: {}", id.id()),
+                    UserOrGroup::Group { id } => format!("Group: {}", id.id()),
+                })
+                .collect::<Vec<_>>();
+            let failures = encrypt_result
+                .access_errs()
+                .iter()
+                .map(|edit_err| match &edit_err.user_or_group {
+                    UserOrGroup::User { id } => {
+                        format!("User: {}, Error: {}", id.id(), edit_err.err)
+                    }
+                    UserOrGroup::Group { id } => {
+                        format!("Group: {}, Error: {}", id.id(), edit_err.err)
+                    }
+                })
+                .collect::<Vec<_>>();
+            println!("Successfully encrypted file to: {:#?}", successes);
+            println!("Failed to encrypt file to: {:#?}", failures);
+            std::fs::write(&output, encrypt_result.encrypted_data())?;
+            println!("Output encrypted file to {:?}", output.display());
+        }
+        CommandLineArgs::FileDecrypt {
+            filename,
+            device_path,
+            output: maybe_output,
+        } => {
+            let output = match maybe_output {
+                // User specified output file
+                Some(desired) => desired,
```

Comment:
maybe_output to filename of decrypted can be its own function.

Reply:
Done and tested